iocage_lib/ioc_plugin.py

```diff
@@ -84,6 +84,125 @@ def __init__(self, release=None, jail=None, plugin=None, branch=None,
             # Backwards compat
             self.branch = 'master'
 
+    def clone_repo(self, depth=None):
+        if self.server == "download.freebsd.org":
+            git_server = "https://github.com/freenas/iocage-ix-plugins.git"
+        else:
+            git_server = self.server
+
+        git_working_dir = f"{self.iocroot}/.plugin_index"
+
+        if os.geteuid() == 0:
+            try:
+                self._clone_repo(
+                    self.branch, git_server, git_working_dir,
+                    depth, self.callback
+                )
+            except Exception as err:
+                iocage_lib.ioc_common.logit(
+                    {
+                        "level": "EXCEPTION",
+                        "message": err
+                    },
+                    _callback=self.callback
+                )
+
+    @staticmethod
+    def fetch_plugin_packagesites(package_sites):
+        def download_parse_packagesite(packagesite_url):
+            package_site_data = {}
+            try:
+                response = requests.get(f'{packagesite_url}/All/')
+                response.raise_for_status()
+
+                for pkg in re.findall(r'<a.*>(.*).txz</a>', response.text):
```

Comment:
These are greedy, perhaps `r'<a.*?>(.*?).txz</a>'`, also wondering if you dont need re.S and re.M? i dont think is guaranteed to have them all on same line.

As even safer approach would be xml parsing.

Reply:
@william-gr i tried xml parsing first, regex is 2x times faster. The way we have it right now, the flags aren't needed, but better safe then sorry, i'll add those ;)

Reply:
Just remembered  we probably dont want another dependency in iocage, so this should be fine :P

Reply:
Perhaps `r'\S+.txz'` ?

Reply:
Yup, the regex flags are not needed in our case. Just a little more to the regex ;)